src/problem_updates.jl

```diff
@@ -33,14 +33,17 @@ update_initial_estimate(prob::OptimizationProblem, sol::SciMLBase.OptimizationSo
     )
 end
 
-@inline function replace_objective_function(prob::F, obj::FF) where {F<:OptimizationProblem,FF}
-    g = replace_objective_function(prob.f, obj)
+@inline function replace_objective_function(prob::F, obj::FF, grad, hess) where {F<:OptimizationProblem, FF}
+    g = replace_objective_function(prob.f, obj, grad, hess)
     return remake(prob; f=g)
 end
 
 # fix the objective function's nth parameter at θₙ
 function construct_fixed_optimisation_function(prob::OptimizationProblem, n::Integer, θₙ, cache)
```

Comment:
You pretty much test this indirectly in Custom_hessian.jl, but this just makes it easier down the line in case something here breaks we can quickly rule out if it's due to an incorrect calculation.

Reply:
Will add test, good unit tests are useful for avoiding problems downstream :)

Reply:
Added a unit test and tested both gradient and Hessian for a three-dimensional model.